client/allocrunner/tasklifecycle/gate.go

```diff
@@ -0,0 +1,80 @@
+package tasklifecycle
+
+const (
+	gateClosed = false
+	gateOpened = true
+)
+
+// Gate is used by the Coordinator to block or allow tasks from running.
+//
+// It provides a channel that taskRunners listens on to determine when they are
+// allowed to run. The Gate has an infinite loop that is either feeding this
+// channel (therefore allowing listeners to proceed) or not doing anything
+// (causing listeners to block an wait).
+//
+// The Coordinator uses the Gate Open() and Close() methods to control this
+// producer loop.
+type Gate struct {
```

Comment:
Calling this `Gate` is a great metaphor and the use of the API is really clear as a result 👍